app/models/sequencescape_submission.rb

```diff
@@ -81,10 +107,16 @@ def asset_groups=(asset_groups)
                     end
   end
 
+  def asset_groups_for_orders_creation
+    return asset_groups unless (asset_groups.length == 1) && extra_barcodes
```

Comment:
I'm interested to know what's the  scenario where asset_groups length is > 1 ?

Reply:
All new assets are added to the same asset group defined, that it should be only 1 when we have extra_barcodes because all assets are added to the same asset group, if we found more than 1 asset group and we have extra_barcodes, then there is something that didnt go well/was not expected